table/block_based/block_based_table_reader_impl.h

```diff
@@ -54,7 +55,12 @@ TBlockIter* BlockBasedTable::NewDataBlockIterator(
   CachableEntry<Block> block;
   s = RetrieveBlock(prefetch_buffer, ro, handle, dict, &block, block_type,
                     get_context, lookup_context, for_compaction,
-                    /* use_cache */ true, /* wait_for_cache */ true);
+                    /* use_cache */ true, /* wait_for_cache */ true,
+                    async_read);
+
+  if (s.IsTryAgain() && async_read) {
+    return iter;
```

Comment:
What is the iter status? Should it also be set to TryAgain()?

Reply:
I tried to set it but it seems like it can't be reset. It can only be invalidated. So the status will be Status::OK.

Reply:
Nvm. The block_iter_ status is ignored if async_read_in_progress_ is true, so it doesn't matter.